### PR TITLE
Add a local TTS service 

### PR DESCRIPTION
**Pull request by emilyxzhou**
The TTS setup script currently has hardcoded directory paths; I'm planning to change this to a relative path to match the other setup scripts (`get_deepspeech_models.sh`, `get_w2l_models.sh`). 

I also noticed that the speaker service is expecting a wav file instead of ogg, which is why I set the TTS service to write the audio data to `tts.wav` instead.

### CODE REVIEW

File: harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+        # Load configs
+        self._tts_config = load_config(tts_config)
+        self._vocoder_config = load_config(vocoder_config)
+
+        # Make sure scale_stats.npy path is correct when using with roslaunch
+        scale_stats_path = os.path.join(
+            "/root", "harmoni_catkin_ws", "src", "HARMONI", "harmoni_actuators",
+            "harmoni_tts", "scale_stats.npy"
+        )
```
Comment (chrismbirmingham) [inline harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py]: The path should either be relative or a parameter.

File: harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        verbose=False,
+        speedup=1.1
+    ):
+        self._use_cuda = use_cuda
+        self._verbose = verbose
+        self._speedup = speedup
+
+        self._check_files_exist([
+            tts_config, tts_model, vocoder_config, vocoder_model
+        ])
```
Comment (chrismbirmingham) [inline harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py:36]: The config folder appeared to be empty for me.

**Review by chrismbirmingham — COMMENTED** [review]
Great work so far, please add setup instructions in the readme and make sure they work from a clean install. I ran into issues with a few imports and config files.

**Comment by emilyxzhou** [comment]
@chrismbirmingham: The readme has been updated with install instructions, and all dependencies have been included in the kinetic and noetic base dockerfiles. I also confirmed that the local TTS service works with a clean install.

File: harmoni_actuators/harmoni_tts/config/tts_local_configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+tts:
+  default_param:
+    tts_config: "/root/harmoni_catkin_ws/model/tts/config.json"
+    tts_model: "/root/harmoni_catkin_ws/model/tts/tts_model.pth.tar"
+    vocoder_config: "/root/harmoni_catkin_ws/model/tts/config_vocoder.json"
+    vocoder_model: "/root/harmoni_catkin_ws/model/tts/vocoder_model.pth.tar"
+    scale_stats_path: "/root/harmoni_catkin_ws/src/HARMONI/harmoni_actuators/harmoni_tts/scale_stats.npy"
```
Comment (RMichaelSwan) [inline harmoni_actuators/harmoni_tts/config/tts_local_configuration.yaml]: To make things cleaner for the local install/non-docker folks and avoid absolute paths, I propose we add a model meta-package which would just serve as an easily findable folder for all AI models, e.g. `$(find harmoni_model)/.../your_model`

File: harmoni_actuators/harmoni_tts/nodes/local_tts_service.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        instance_id = rospy.get_param("instance_id")
+        name = ActuatorNameSpace.tts.name + "_" + instance_id
+        self.speaker_action_client = HarmoniActionClient(name)
+
+        """Setup the TTS service as server"""
+        self.state = State.INIT
+        return
+
+    def request(self, input_text):
+        """[summary]
```
Comment (RMichaelSwan) [inline harmoni_actuators/harmoni_tts/nodes/local_tts_service.py]: missing summary.

File: harmoni_actuators/harmoni_tts/nodes/local_tts_service.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        self.speaker_action_client.send_goal(
+            action_goal=ActionType.DO.value,
+            optional_data=file_path,
+            wait=False
+        )
+
+
+def main():
+    """[summary]
+    Main function for starting local TTS service
```
Comment (RMichaelSwan) [inline harmoni_actuators/harmoni_tts/nodes/local_tts_service.py]:
```suggestion
    Main function for starting local TTS service
```

File: harmoni_actuators/harmoni_tts/config/tts_local_configuration.yaml (hunk 3 above)
Comment (emilyxzhou) [reply harmoni_actuators/harmoni_tts/config/tts_local_configuration.yaml]: That sounds like a good solution to the path issue. Would it be better to add that in this PR and update the other 2 services afterwards, or start a new PR later?

File: harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py (hunk 2 above)
Comment (micolspitale93) [reply harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py:36]: The same. It doesn't exist the file config.json in the model path.

**Review by micolspitale93 — COMMENTED** [review]
I was not able to test the service because of an "invalid path error" . You could also include the TTS folder into the gitignore

File: harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py (hunk 2 above)
Comment (emilyxzhou) [reply harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py:36]: @micolspitale93, did you first run `sh harmoni_actuators/harmoni_tts/setup_tts.sh` from the HARMONI directory? This script will download the models and config files.

Comment (micolspitale93) [reply harmoni_actuators/harmoni_tts/src/harmoni_tts/local_tts_client.py:36]: My bad, I'll try it.

File: harmoni_actuators/harmoni_tts/config/tts_local_configuration.yaml (hunk 3 above)
Comment (RMichaelSwan) [reply harmoni_actuators/harmoni_tts/config/tts_local_configuration.yaml]: I would do it the best way in any unmerged code (this PR) and then do another issue/PR for anything that is still doing it the "old" way. That said, if you think it's more efficient/cleaner, feel free to make a new issue/PR to do it all at once later.

**Review by RMichaelSwan — APPROVED** [review]
Looks good to me. Assuming @micolspitale93 was able to test it successfully, I think we are ready to merge.

**Comment by micolspitale93** [comment]
Good to me as well. Ready to merge 👍

**Pull request merged**